pyccel/ast/core.py

```diff
@@ -3149,90 +3149,135 @@ def __init__(
 
     @property
     def name(self):
+        """ Name of the function """
         return self._name
 
     @property
     def arguments(self):
+        """ List of variables which are the function arguments """
         return self._arguments
 
     @property
     def results(self):
+        """ List of variables which are the function results """
         return self._results
 
     @property
     def body(self):
+        """ CodeBlock containing all the statements in the function """
         return self._body
 
     @property
     def local_vars(self):
+        """ List of variables defined in the function """
         return self._local_vars
 
     @property
     def global_vars(self):
+        """ List of global variables used in the function """
         return self._global_vars
 
     @property
     def cls_name(self):
+        """ String containing the name of the class to which the method belongs.
+        If the function is not a class procedure then this returns None """
         return self._cls_name
 
     @property
     def hide(self):
+        """ False if function is pyccelized, True otherwise """
         return self._hide
 
     @property
     def kind(self):
+        """ Returns function or procedure """
         return self._kind
 
     @property
     def imports(self):
+        """ List of imports in the function """
         return self._imports
 
     @property
     def decorators(self):
+        """ List of decorators applied to the function """
         return self._decorators
 
     @property
     def headers(self):
+        """ List of headers applied to the function """
         return self._headers
 
     @property
     def templates(self):
+        """ List of templates used to determine the types """
         return self._templates
```

Comment:
@rmahjoubi Is this docstring correct? Why do we need `templates` in a `FunctionDef`? It would make sense in an `Interface` but I don't understand the relevance here

Reply:
Ah I see the problem! It definitely feels odd to me but I suppose it makes sense. A `FunctionDef` represents a python function before the semantic stage so it doesn't need to have a fixed type and be split into `Interface`s yet.

If we think of it in this way does it make more sense to access the templates via the decorators and headers rather than having it be an integral part of what a `FunctionDef` is?

Reply:
That's definitely better, so an issue for this ?

Reply:
Yes please :)